### DISCUSSION TITLE
How to listen to other configuration files

### DISCUSSION THREAD

**Discussion by kurorinto** [web-infra-dev/rsbuild#3021] [Q&A] [answered]
```bash
├── configs
│   ├── configA.ts
│   └── configB.ts
├── src
└── rsbuild.config.ts
```
rsbuild config
```typescript
import configA from './configs/configA';
import configB from './configs/configA';

export default defineConfig(async () => {
  return {
    ['propA']: configA,
    ['propB']: configB,
    // some general configs
  };
});
```
How to listen for 'configs' folder or 'configA/B.ts' file to change and re-run `rsbuild dev` or `rsbuild.startDevServer()`

**Comment by liyincode** [comment, marked as answer]
[dev.watchFiles](https://rsbuild.dev/config/dev/watch-files)

in `rsbuild.config.ts`
```
export default defineConfig({
  dev: {
    watchFiles: {
      type: 'reload-server',
      paths: ['./configs/configA.ts', './configs/configB.ts'] 
    }
  }
});
```
You can use this configuration to listen to the configA.ts and configB.ts files separately
Or use `paths: ['. /configs/*.ts'] ` listen on all ts files in the configs directory

Any change in the contents of the file triggers a devServer restart

**Reply by kurorinto** [reply]
thanks!
